### PR TITLE
[SPARK-49199][BUILD] `build/mvn` should check project local installed Maven ahead

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

This PR modifies the `mvn` command detection logic in `build/mvn`, to check project local installed `build/apache-maven-${MVN_VERSION}/bin/mvn` ahead to avoid unnecessary network accessing.

### Why are the changes needed?

To speed up `build/mvn` bootstrap time.

I found sometimes `build/mvn -version` is pretty slow even `build/apache-maven-${MVN_VERSION}` is already cached, and I found it may stuck at the following call when network quality is not good.

https://github.com/apache/spark/blob/879924491944be5d022df895b947bcee25837cc0/build/mvn#L129

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Tested `build/mvn -version` in the following cases:
1. when `build/apache-maven-${MVN_VERSION}/bin/mvn` is available, the command always finishes quickly.
2. when `build/apache-maven-${MVN_VERSION}/bin/mvn` is unavailable but global installed `mvn` version matches `maven.version` defined in root `pom.xml`, the command always finishes quickly too.
3. otherwise, it automatically downloads the maven binary tarball from network.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by pan3793** [comment]
To reviewers, it's recommended to hide whitespace on viewing the diff of this patch
image

File: build/mvn [hunk 1]
```diff
@@ -115,34 +110,37 @@ function version { echo "$@" | awk -F. '{ printf("%03d%03d%03d\n", $1,$2,$3); }'
...
-    local MIRROR_URL_QUERY="?action=download"
-
-    if [ $(command -v curl) ]; then
-      if ! curl -L --output /dev/null --silent --head --fail "${APACHE_MIRROR}/${FILE_PATH}${MIRROR_URL_QUERY}" ; then
-        # Fall back to archive.apache.org for older Maven
-        echo "Falling back to archive.apache.org to download Maven"
-        APACHE_MIRROR="https://archive.apache.org/dist"
-        MIRROR_URL_QUERY=""
-      fi
+  MVN_BIN="${_DIR}/apache-maven-${MVN_VERSION}/bin/mvn"
```
[the remark this answers is not in the document]
Comment (pan3793) [reply build/mvn:113]: this also works, changed as suggested

**Review by LuciferYang — APPROVED** [review]

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please recover `install_app` because we still need to verify the checkable binary after untarring the downloaded tarball, @pan3793 .

**Comment by pan3793** [comment]
@dongjoon-hyun okay, restored.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @pan3793 and @LuciferYang .
Merged to master.

**Pull request closed without merging**